ui/ducks/send/send.js

```diff
@@ -1326,6 +1326,7 @@ export function toggleSendMaxMode() {
       await dispatch(actions.updateAmountMode(AMOUNT_MODES.MAX));
       await dispatch(actions.updateAmountToMax());
     }
+    await dispatch(computeEstimatedGasLimit());
```

Comment:
@danjm Do we want to do this for non-token sends?

Reply:
Yes, in case custom networks like optimism need it

Reply:
We could not do it for non-token sends on mainnet

Reply:
hmmm yeah maybe... Looking a bit more through the send slice, it looks like in the updateSendAmount action creator we are not recomputing the gas limit for non-token assets when the amount is updated, which as you point out for custom networks (like optimism) may cause an issue?

Reply:
^^^ https://github.com/MetaMask/metamask-extension/blob/4e87069af3619db2307e08e1d577621339eab3cc/ui/ducks/send/send.js#L1167

Reply:
@danjm this isn't a blocker. Just raising it here because you've pointed out that we shouldn't use the asset_type to decide whether or not to recompute gas

Reply:
right, and that could also matter if the user was sending to a contract address that is something other than a token

Reply:
Should I write up a ticket for this?